source/ui/BioXAS/BioXASWigglerView.cpp

```diff
@@ -0,0 +1,62 @@
+#include "BioXASWigglerView.h"
+
+BioXASWigglerView::BioXASWigglerView(BioXASWiggler *wiggler, QWidget *parent) :
+    QWidget(parent)
+{
+	wiggler_ = 0;
+
+	// Create UI elements.
+
+	gapEditor_ = new CLSControlEditor(0);
+	gapEditor_->setTitle("Gap");
+	gapEditor_->setMinimumWidth(200);
+
+	gapStatusEditor_ = new CLSControlEditor(0);
+	gapStatusEditor_->setTitle("Gap status");
+	gapStatusEditor_->setMinimumWidth(200);
+
+	// Create and set main layouts.
+
+	QVBoxLayout *layout = new QVBoxLayout();
+	layout->setMargin(0);
+	layout->addWidget(gapEditor_);
+	layout->addWidget(gapStatusEditor_);
+
+	setLayout(layout);
+
+	// Current settings.
+
+	setWiggler(wiggler);
+}
+
+BioXASWigglerView::~BioXASWigglerView()
+{
+
+}
+
+void BioXASWigglerView::setWiggler(BioXASWiggler *newWiggler)
+{
+	if (wiggler_ != newWiggler) {
+		wiggler_ = newWiggler;
+		emit wigglerChanged(wiggler_);
+	}
+
+	updateGapEditor();
+	updateGapStatusEditor();
```

Comment:
The editors need to be updated as part of the first call to `setWiggler(...)` in the constructor. They need to be updated even if the wiggler instance hasn't changed.